### DISCUSSION TITLE
Where is the immediate cache stored ?

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [ImageMagick/ImageMagick#3166] [Help] [answered]
Hello, I need to resize 6k+ images from around 4k to 2k size in pixels and I am using PHP library as a wrapper on a ubuntu server.

I did set up some big numbers in `policy.xml` with automatic temp folder clearing while script runs, but when it IS running, it eats so much of my disk and I can't find the files that are being "left over" while it's running for me to manually remove them.

Once I kill it, the whatever imagemagick was doing frees up the space.

Where is imagemagick storing it's things, that I can not find them anywhere and why is imagemagick not clearing them, only once it's killed or done converting ?

It makes converting many images impossible, because I don't have infinite disk to allocate as a "cache". I need it to use, lets say 4gigs and once it tries to go over, clear old cache or something.

Any tips ?

**Comment by snibgo** [comment, marked as answer]
What version of IM?

What is your exact IM command? Perhaps you are trying to read all your 6000 images into memory, then resizing them. If so, then I suggest using "mogrify" so only one image at a time is in memory.

**Reply by FluffyDiscord** [reply]
Server has V6.

I fiddled around with it for a bit and it was leaving traces in memory/disk because I was using once instance of imagemagick to convert all images at once, just like as you said, so the whatever thing it was leaving behind could not be cleaned, because the process reference was still active.

I went ahead and make once instance per image and now its fine.

Can I somehow mark this question as solved/answered ?

**Comment by snibgo** [comment]
> Can I somehow mark this question as solved/answered ?

Yes. There is a button or something for that.
